src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -51,11 +51,23 @@
 #include "utils/errcodes.h"
 #include "utils/lsyscache.h"
 #include "utils/rel.h"
-#include "utils/relcache.h"
-#include "utils/typcache.h"
+
+#include "catalog/pg_proc.h"
+#include "optimizer/planmain.h"
+
+
+typedef struct {
+	bool containsVar;
+	bool varArgument;
+	bool badCoalesce;
+} WalkerState;
 
 
 /* planner functions forward declarations */
+static bool ContainsDisallowedFunctionCalls(Node *expression, bool *varArgument,
```

Comment:
I'd like to rename that function to something that makes it clearer that it's about functioncalls for DML, not generally forbidden functions.